tools/android_lint/bin/main.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+
+Future<void> main(List<String> args) async {
+  final ArgParser argParser = ArgParser();
+  argParser.addOption(
+    'in',
+    help: 'The path to `engine/src`.',
+    defaultsTo: path.join(path.current),
+  );
+  argParser.addOption(
+    'out',
+    help: 'The path to write the generated the HTML report to.',
+    defaultsTo: 'lint_report',
+  );
+
+  final ArgResults argResults = argParser.parse(args);
+  final Directory androidDir = Directory(path.join(
+    argResults['in'],
+    'flutter',
+    'shell',
+    'platform',
+    'android',
+  ));
+  if (!androidDir.existsSync()) {
+    print('This command must be run from the engine/src directory, '
+        'or be passed that directory as the --in parameter.\n');
+    print(argParser.usage);
+    exit(-1);
+  }
+
+  final Directory androidSdkDir = Directory(
+    path.join(argResults['in'], 'third_party', 'android_tools', 'sdk'),
+  );
+
+  if (!androidSdkDir.existsSync()) {
+    print('The Android SDK for this engine is missing from the '
+        'third_party/android_tools directory. Have you run gclient sync?\n');
+    print(argParser.usage);
+    exit(-1);
+  }
+
+  final IOSink projectXml = File('./project.xml').openWrite();
+  projectXml.write('''<!-- THIS FILE IS GENERATED. PLEASE USE THE INCLUDED DART PROGRAM  WHICH -->
```

Comment:
I like that this is both auto generating this and committing it to source, means that we don't need to fiddle with a config file manually but if something goes wrong with automatically finding the files to include that'll be obvious when it's time to commit. Should we do anything here to stop this from dynamically changing `project.xml` in CI though? Like only running this based on an additional arg to rewrite it first. Thinking about potential bugs where it can't find any Java files on a run for some weird filesystem reason and then reports 0 errors found in all 0 files.

Reply:
I think it should be fine for CI - I'd be more worried about someone forgetting to re-run it after adding a new java file than it messing up.  I suspect if there were a FS related issue it should cause an exception and fail this.